### ISSUE TITLE
Optimize EKO in flavor

### ISSUE THREAD

**Issue by alecandido** [NNPDF/eko#99]
We can optimize EKOs final object for cases in which it's known that not all flavors are needed.

This is corresponding to a similar optimization done at the level of PineAPPL `FkTables` https://github.com/N3PDF/pineappl/issues/79.

**Comment by felixhekhorn** [comment]
(With the current layout,) actually, there is nothing to do then calling `flavor_reshape` ...

**Comment by AleCandido** [comment]
I agree, but we can provide a wrapper to generate a suitable evolution basis, just specifying how many flavors you want (`FkAssumptions`-like).
